fairseq/models/fairseq_model.py

```diff
@@ -68,6 +68,14 @@ def load_state_dict(self, state_dict, strict=True, args=None):
         """
         self.upgrade_state_dict(state_dict)
         new_state_dict = prune_state_dict(state_dict, args)
+
+        # Since layers are now pruned, *_layers_to_keep are no longer needed.
+        # This is more of "It would make it work fix" rather than a proper fix.
+        if "encoder_layers_to_keep" in vars(args):
+            args.encoder_layers_to_keep = None
+        if "decoder_layers_to_keep" in vars(args):
+            args.decoder_layers_to_keep = None
+
```

Comment:
Please move this to the end of prune_state_dict instead: https://github.com/pytorch/fairseq/blame/master/fairseq/checkpoint_utils.py#L397